operators/src/plot/histogram.rs

```diff
@@ -0,0 +1,841 @@
+use geoengine_datatypes::primitives::{
+    DataRef, FeatureDataRef, FeatureDataType, Geometry, Measurement,
+};
+use geoengine_datatypes::raster::{Pixel, RasterTile2D};
+use serde::{Deserialize, Serialize};
+use snafu::{ensure, OptionExt};
+use std::convert::TryFrom;
+
+pub const HISTOGRAM_OPERATOR_NAME: &str = "Histogram";
+
+/// A histogram plot about either a raster or a vector input.
+///
+/// For vector inputs, it calculates the histogram on one of its attributes.
+///
+pub type Histogram = Operator<HistogramParams>;
+
+/// The parameter spec for `Histogram`
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct HistogramParams {
+    /// Name of the (numeric) attribute to compute the histogram on. Ignored for operation on rasters.
+    pub column_name: Option<String>,
+    /// The bounds (min/max) of the histogram.
+    pub bounds: HistogramBounds,
+    /// If the bounds are empty, it is derived from the square-root choice rule.
+    pub buckets: Option<usize>,
+    /// Whether to create an interactive output (`false` by default)
+    #[serde(default)]
+    pub interactive: bool,
+}
+
+string_token!(Data, "data");
+
+/// Let the bounds either be computed or given.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum HistogramBounds {
+    Data(Data),
+    Values { min: f64, max: f64 },
+    // TODO: use bounds in measurement if they are available
+}
+
+#[typetag::serde]
+impl PlotOperator for Histogram {
+    fn initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<InitializedPlotOperator>> {
+        ensure!(
+            self.vector_sources.len() + self.raster_sources.len() == 1,
+            error::InvalidNumberOfVectorInputs {
+                expected: 1..2,
+                found: self.vector_sources.len() + self.raster_sources.len()
+            }
+        );
+
+        let vector_sources = self
+            .vector_sources
+            .into_iter()
+            .map(|o| o.initialize(context))
+            .collect::<Result<Vec<_>>>()?;
+        if !vector_sources.is_empty() {
+            let column_name =
+                self.params
+                    .column_name
+                    .as_ref()
+                    .context(error::InvalidOperatorSpec {
+                        reason: "Histogram on vector input is missing `column_name` field"
+                            .to_string(),
+                    })?;
+
+            let vector_result_descriptor = vector_sources[0].result_descriptor();
+
+            match vector_result_descriptor.columns.get(column_name) {
+                None => {
+                    return Err(Error::ColumnDoesNotExist {
+                        column: column_name.to_string(),
+                    });
+                }
+                Some(FeatureDataType::Categorical) | Some(FeatureDataType::Text) => {
+                    // TODO: incorporate categorical data
```

Comment:
histograms on strings columns are also interesting. Of course only if there are few different values and we might consider that Categorical data, but then we need a convenient way for users to use the operator that way..

Reply:
I guess that is a different task, since it is not easy to to it for strings, e.g., if each feature has a different string, you'll get into memory problems.